lib/Controller/LoginController.php

```diff
@@ -344,6 +353,10 @@ public function code($state = '', $code = '', $scope = '', $error = '', $error_d
 		$this->userSession->createSessionToken($this->request, $user->getUID(), $user->getUID());
 		$this->userSession->createRememberMeToken($user);
 
+		// for backchannel logout
+		$this->config->setAppValue(Application::APP_ID, 'sid-' . $idTokenPayload->sid, $this->session->getId());
```

Comment:
Generally IConfig might have the issue that there is no proper way that we can expire this, so if users don't logout manually this will be growing forever. Maybe we should go with either caching like redis or a dedicated db table?

Reply:
Caching is not an option here because we need to be sure the value is stored and caching might not have been configured in the instance.

I thought I could find a trick to cleanup app config values but you're right, it might not be possible.

Let's go for a dedicated table then. The cleanup mechanism would then be a cron job deleting sid/sessionId links that are older than...one day? What do you think?

Reply:
Or a smarter way to clean things up is to delete `oc_user_oidc_sessions` (the future table) entries referring to entries that do not exists anymore in `oc_authtoken`.